src/test/java/com/cdancy/jenkins/rest/features/JobsApiLiveTest.java

```diff
@@ -112,12 +123,31 @@ public void testLastBuildNumberOnJob() {
         assertTrue(buildNumber == 1);
     }
 
+
+    @Test(dependsOnMethods = "testBuildJob")
+    public void testLastStableBuildNumberOnJob() {
+        try {
+            Thread.sleep(10000);
```

Comment:
Waiting in a loop is brittle. Instead I suggest you poll until Jenkins reaches the state needed for the test. Here is an example of waiting for the build to start: https://github.com/cdancy/jenkins-rest/blob/359dd0461330e6f620caa9981550f3e4ba5ce5f2/src/test/java/com/cdancy/jenkins/rest/BaseJenkinsApiLiveTest.java#L83
If you need for the build to be completed, the wiki [shows an example](https://github.com/cdancy/jenkins-rest/wiki/RunningABuild).